### DISCUSSION TITLE
如何使用全局css样式表文件

### DISCUSSION THREAD

**Discussion by FalconWu2017** [ofajs/discussion#19] [General]
如何一次加载css文件，对全部组件内的元素都有效？比如我想网站全部 button、input、lable等统一样式。

**Comment by kirakiray** [comment]
这个方案还在建设中，因为引入global style 的同时，会污染第三方组件的代码，导致更难排查的问题；而且容易滥用，导致类似`window`全局作用域的问题。

建议你建立一个 `public.css`，这个 css 文件内使用 [@import](https://developer.mozilla.org/zh-CN/docs/Web/CSS/@import) 语法引用共用样式文件，然后你的所有相关组件共用这个 public.css 资源，就能达到你的需求。（这个管理资源的做法其实就像js的import 模块化）

如果你有好的方案，可以在这里回复提供解决方案：https://github.com/kirakiray/ofa.js/issues/140

**Reply by FalconWu2017** [reply]
其实我是做后端开发的，对前端也不是非常了解，以前只用vue和原生html js css写几个管理型页面。  
我用vue和原生页面的时候只要css加载了就对整个页面就有效了，不知道为什么ofa不会生效。

**Reply by kirakiray** [reply]
👌 近期应该会更新一版关于 globalCSS 的特征。

**Comment by kirakiray** [comment]
已添加 `global-link` 组件，你可以查阅相关文档进行使用。使用前确保 ofa.js 版本大于等于 4.5.23。

* 使用方法：https://github.com/kirakiray/ofa.js/blob/main/libs/global-link
* 测试代码：https://github.com/kirakiray/ofa.js/blob/main/libs/global-link/test/test.html
* 测试代码在线预览：https://kirakiray.github.io/ofa.js/libs/global-link/test/test.html
